src/AutoUI/Filters/Filters.tsx

```diff
@@ -96,6 +106,182 @@ const removeFieldsWithNoFilter = (schema: JSONSchema): JSONSchema => {
 	return schema;
 };
 
+/**
+ * Constructs a schema or modifies properties of an existing schema based on a reference scheme path.
+ * This function recursively applies changes to nested properties or items based on the transformed reference scheme.
+ *
+ * @param schemaOrProperties - The schema or the properties part of a schema to modify.
+ * @param transformedXRefScheme - An array of property names extracted from a reference scheme, indicating the path through the schema.
+ * @param description - Optional new description to apply to the schema at the specified path.
+ * @param title - Optional new title to apply to the schema at the specified path; defaults to existing title if not provided.
+ * @returns A new schema definition with modifications applied as per the reference scheme path.
+ */
+const constructSchemaProperties = (
+	schemaOrProperties: JSONSchemaDefinition,
+	transformedXRefScheme: string[] | undefined,
+	description?: string,
+	title?: string,
+): JSONSchemaDefinition => {
+	// Return the original schema if there is nothing to transform or no path is provided.
+	if (
+		schemaOrProperties == null ||
+		!transformedXRefScheme?.length ||
+		!isJSONSchema(schemaOrProperties)
+	) {
+		return schemaOrProperties;
+	}
+
+	// Deconstruct the transformedXRefScheme into the first key and the rest of the path.
+	const [firstRefSchemeKey, ...restRefScheme] = transformedXRefScheme;
+
+	// If the current level is a JSON schema and has properties, recursively apply the transformation to these properties.
+	if (schemaOrProperties.properties) {
+		return {
+			...schemaOrProperties,
+			title: title ?? schemaOrProperties.title, // Use provided title or default to existing title.
+			description, // Apply the new description.
+			properties: constructSchemaProperties(
+				schemaOrProperties.properties,
+				restRefScheme,
+				description,
+			) as JSONSchema['properties'], // Recurse into properties.
+		};
+	}
+	// If the current level is a JSON schema and has an items definition, apply the transformation to the items.
+	if (schemaOrProperties.items) {
+		return {
+			...schemaOrProperties,
+			title: title ?? schemaOrProperties.title, // Use provided title or default to existing title.
+			description, // Apply the new description.
+			items: constructSchemaProperties(
+				schemaOrProperties.items as JSONSchema,
+				restRefScheme,
+				description,
+			), // Recurse into items.
+		};
+	}
```

Comment:
```suggestion
	if (schemaOrProperties.properties || schemaOrProperties.items) {
		return {
			...schemaOrProperties,
			title: title ?? schemaOrProperties.title,
			description,
			properties: constructSchemaProperties(
				schemaOrProperties.properties ?? (schemaOrProperties.items as JSONSchema),
				restRefScheme,
				description,
			) as JSONSchema['properties'],
		};
	}
```
(keep comments of course, I just removed them for the sake of showing purely the suggestion)

Reply:
I understand the point but we need to change properties and items in the returned object. Also, the as is different as 1 represent JSONSchema['properties'] and the other JSONSchema['items'].

🤔 we can for sure reduce the code, I just would like to avoid adding complexity. Let me give it a try

Reply:
we could do something like this:

```
  if (schemaOrProperties.properties || schemaOrProperties.items) {
    const properties = schemaOrProperties.properties
      ? (constructSchemaProperties(
          schemaOrProperties.properties,
          restRefScheme,
          description
        ) as JSONSchema["properties"])
      : undefined;
    const items: JSONSchema["items"] = schemaOrProperties.items
      ? constructSchemaProperties(
          schemaOrProperties.items as JSONSchema,
          restRefScheme,
          description
        )
      : undefined;
    return {
      ...schemaOrProperties,
      title: title ?? schemaOrProperties.title, // Use provided title or default to existing title.
      description, // Apply the new description.
      properties, // Recurse into properties.
      items, // Recurse into items.
    };
  }
```

WDYT ? would you find it more readable or nicer ?

Reply:
I don't like the redundancy. Also I realized I made a mistake in my earlier suggestion, fixed here, though we need to do the cast:
```suggestion
	if (schemaOrProperties.properties || schemaOrProperties.items) {
		return {
			...schemaOrProperties,
			title: title ?? schemaOrProperties.title,
			description,
			[schemaOrProperties.properties ? 'properties' : 'items']: constructSchemaProperties(
				schemaOrProperties.properties ?? (schemaOrProperties.items as JSONSchema),
				restRefScheme,
				description,
			),
		};
	}
```
> I just would like to avoid adding complexity

This doesn't really feel complex IMO 😓 What do you find complex about it?

> WDYT ? would you find it more readable or nicer ?

If my suggestion doesn't work, this is better than nothing